code/modules/clothing/chameleon.dm

```diff
@@ -616,15 +653,22 @@
 	chameleon_action.emp_randomise(INFINITY)
 
 /obj/item/clothing/mask/chameleon/attack_self(mob/user)
-	vchange = !vchange
-	to_chat(user, "<span class='notice'>The voice changer is now [vchange ? "on" : "off"]!</span>")
+	voice_change = !voice_change
+	to_chat(user, "<span class='notice'>The voice changer is now [voice_change ? "on" : "off"]!</span>")
 
+/obj/item/clothing/mask/chameleon/get_name(mob/user, var/default_name)
+	var/mob/living/carbon/human/H = user
+	if(voice_change && H.wear_id)
+		var/obj/item/card/id/idcard = H.wear_id.GetID()
+		if(istype(idcard) && idcard.electric)
+			default_name = idcard.registered_name
```

Comment:
```suggestion
			return idcard.registered_name
```
Technically it does work before, but that syntax typically isn't liked in a lot of languages and can cause confusion as it is unclear if this is a local variable or not at a first glance.